test/opt/copy_prop_array_test.cpp

```diff
@@ -1884,6 +1884,81 @@ OpFunctionEnd
   SetTargetEnv(SPV_ENV_UNIVERSAL_1_4);
   SinglePassRunAndMatch<CopyPropagateArrays>(before, false);
 }
+
+TEST_F(CopyPropArrayPassTest, StructIndexCannotBecomeDynamic) {
```

Comment:
Generally looks good.  However, this test is a bit long.  Could you try to reduce it?  For example, you should only need a single function.  You could run the inliner to do that for you.  The OpName instructions are not needed. There is a tool (`spirv-reduce`) that could help.

Reply:
Done, thanks for pointing this out!